Operations fail on quatslice of a transposed 4d array

The following code generates compile time error:
```c++
blaze::DynamicArray<4UL, int> q{{{{{1, 2}, {3, 4}},
                                          {{5, 6}, {7, 8}},
                                          {{9, 10}, {11, 12}}},
                                         {{{13, 14}, {15, 16}},
                                          {{17, 18}, {19, 20}},
                                          {{21, 22}, {23, 24}}}}};
blaze::DynamicTensor<int> t{{{-5, -3}, {-4, 1}, {0, 2}},
                                    {{-5, -3}, {-4, 1}, {0, 2}}};
blaze::quatslice(blaze::trans(q, {3, 0, 1, 2}), 0) % t;
```
```console
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(171): error C2794: 'OppositeType': is not a member of any direct or indirect base class of 'blaze::INVALID_TYPE'
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(171): note: see reference to alias template instantiation 'blaze::OppositeType_t<blaze::INVALID_TYPE>' being compiled
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(1205): note: see reference to class template instantiation 'blaze::DTensDTensSchurExpr<blaze::QuatSlice<AT>,blaze::DynamicTensor<ET>>' being compiled
1>        with
1>        [
1>            AT=blaze::DQuatTransExpr<blaze::DynamicArray<4,int>>,
1>            ET=int
1>        ]
1>c:\repos\blaze_app_1\prog.cpp(2060): note: see reference to function template instantiation 'decltype(auto) blaze::operator %<blaze::QuatSlice<AT>,blaze::DynamicTensor<ET>>(const blaze::DenseTensor<blaze::QuatSlice<AT>> &,const blaze::DenseTensor<blaze::DynamicTensor<ET>> &)' being compiled
1>        with
1>        [
1>            AT=blaze::DQuatTransExpr<blaze::DynamicArray<4,int>>,
1>            ET=int
1>        ]
1>c:\repos\hpx\libs\parallel_executors\include\hpx\parallel\execution_policy.hpp(648): note: see reference to class template instantiation 'hpx::parallel::execution::parallel_policy_executor<hpx::launch>' being compiled
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(171): error C2938: 'blaze::OppositeType_t<blaze::INVALID_TYPE>' : Failed to specialize alias template
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(172): error C2794: 'TransposeType': is not a member of any direct or indirect base class of 'blaze::INVALID_TYPE'
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(172): note: see reference to alias template instantiation 'blaze::TransposeType_t<blaze::INVALID_TYPE>' being compiled
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(172): error C2938: 'blaze::TransposeType_t<blaze::INVALID_TYPE>' : Failed to specialize alias template
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(173): error C2794: 'ElementType': is not a member of any direct or indirect base class of 'blaze::INVALID_TYPE'
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(173): note: see reference to alias template instantiation 'blaze::ElementType_t<blaze::INVALID_TYPE>' being compiled
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(173): error C2938: 'blaze::ElementType_t<blaze::INVALID_TYPE>' : Failed to specialize alias template
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(176): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(176): error C2144: syntax error: 'unknown-type' should be preceded by ';'
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(176): error C2208: 'unknown-type': no members defined using this type
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(449): error C2955: 'blaze::SIMDTrait': use of class template requires template argument list
1>c:\repos\vcpkg\installed\x64-windows\include\blaze\math\simd\simdtrait.h(296): note: see declaration of 'blaze::SIMDTrait'
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(449): error C2039: 'size': is not a member of 'blaze::SIMDTrait<T>'
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(449): error C2440: 'initializing': cannot convert from 'overloaded-function' to 'const size_t'
1>c:\repos\blaze_tensor\blaze_tensor\math\expressions\dtensdtensschurexpr.h(449): note: Context does not allow for disambiguation of overloaded function
1>Done building project "app1.vcxproj" -- FAILED.
```
If we didn't need to transpose `q` this would have successfully compiled.
Note that `blaze::quatslice(blaze::trans(q, {3, 0, 1, 2}), 0) ` is equivalent to this:
```
((           1            3 ) (           5            7 ) (           9           11 ) )
((          13           15 ) (          17           19 ) (          21           23 ) )
```

--- a/blaze_tensor/math/expressions/DQuatTransExpr.h
+++ b/blaze_tensor/math/expressions/DQuatTransExpr.h
@@ -81,6 +81,7 @@
 #include <blaze_tensor/math/expressions/DQuatTransExprData.h>
 #include <blaze_tensor/math/expressions/DQuatTransposer.h>
 #include <blaze_tensor/math/expressions/QuatTransExpr.h>
+#include <blaze_tensor/math/traits/QuatSliceTrait.h>
 
 
 namespace blaze {
@@ -1134,6 +1135,24 @@ inline decltype(auto) trans( const DArrScalarMultExpr<MT,ST>& dm, RTAs... args )
 /*! \endcond */
 //*************************************************************************************************
 
+
+
+
+//=================================================================================================
+//
+//  QUATSLICETRAIT SPECIALIZATIONS
+//
+//=================================================================================================
+
+//*************************************************************************************************
+/*! \cond BLAZE_INTERNAL */
+template< typename ET, size_t... CTAs, size_t I >
+struct QuatSliceTraitEval2< DQuatTransExpr<ET, CTAs...>, I >
+{
+   using Type = QuatSliceTrait_t< ET >;
+};
+/*! \endcond */
+//*************************************************************************************************
 
 
 

